module/userProjects/helper.js

```diff
@@ -1048,323 +1213,352 @@ module.exports = class UserProjectsHelper {
-            //                  userProfileData.data &&
-            //                  userProfileData.data.response
-            //             ) {
-            //                     projectCreation.data.userProfile = userProfileData.data.response;
-            //                     addReportInfoToSolution = true; 
-            //             } 
-            //         }
+                            let userProfile = await userService.profile(userToken, userId);
+                            if ( userProfile.success && 
+                                 userProfile.data &&
+                                 userProfile.data.response
+                            ) {
+                                    projectCreation.data.userProfile = userProfile.data.response;
+                                    addReportInfoToSolution = true; 
+                            } 
+                        }
 
-            //         projectCreation.data.userRoleInformation = userRoleInformation;
+                    } else {
+                        //Fetch user profile information by calling sunbird's user read api.
+
+                        let userProfileData = await userService.profile(userToken, userId);
+                        if ( userProfileData.success && 
+                             userProfileData.data &&
+                             userProfileData.data.response
+                        ) {
+                                projectCreation.data.userProfile = userProfileData.data.response;
+                                addReportInfoToSolution = true; 
+                        } 
+                    }
                     
-            //         //compare & update userProfile with userRoleInformation
-            //         if ( projectCreation.data.userProfile && userRoleInformation && Object.keys(userRoleInformation).length > 0 && Object.keys(projectCreation.data.userProfile).length > 0 ) {
-            //             let updatedUserProfile = await _updateUserProfileBasedOnUserRoleInfo(
-            //                 projectCreation.data.userProfile,
-            //                 userRoleInformation
-            //             );
-
-            //             if (updatedUserProfile && updatedUserProfile.success == true && updatedUserProfile.profileMismatchFound == true) {
-            //                 projectCreation.data.userProfile = updatedUserProfile.data;
-            //             }
-            //         }
+                    projectCreation.data.userRoleInformation = userRoleInformation;
+                    
+                    //compare & update userProfile with userRoleInformation
+                    if ( projectCreation.data.userProfile && userRoleInformation && Object.keys(userRoleInformation).length > 0 && Object.keys(projectCreation.data.userProfile).length > 0 ) {
+                        let updatedUserProfile = await _updateUserProfileBasedOnUserRoleInfo(
```

Comment:
@Prajwal17Tunerlabs you can rewrite this function because the eleavte user service will have different format

Reply:
cannot find this function in elevate user service